lading/src/blackhole/common.rs

```diff
@@ -0,0 +1,95 @@
+use bytes::Bytes;
+use http_body_util::combinators::BoxBody;
+use hyper::service::Service;
+use hyper_util::{
+    rt::{TokioExecutor, TokioIo},
+    server::conn::auto,
+};
+use lading_signal::Watcher;
+use std::{net::SocketAddr, sync::Arc};
+use tokio::{net::TcpListener, pin, sync::Semaphore, task::JoinSet};
+use tracing::{debug, error, info};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    /// Wrapper for [`std::io::Error`].
+    #[error("IO error: {0}")]
+    Io(std::io::Error),
+}
+
+pub(crate) async fn run_httpd<SF, S>(
+    addr: SocketAddr,
+    concurrency_limit: usize,
+    shutdown: Watcher,
+    make_service: SF,
+) -> Result<(), Error>
+where
+    // "service factory"
+    SF: Send + Sync + 'static + Clone + Fn() -> S,
+    // The bounds on `S` per
+    // https://docs.rs/hyper/latest/hyper/service/trait.Service.html and then
+    // made concrete per
+    // https://docs.rs/hyper-util/latest/hyper_util/server/conn/auto/struct.Builder.html#method.serve_connection.
```

Comment:
```suggestion
    // https://docs.rs/hyper-util/latest/hyper_util/server/conn/auto/struct.Builder.html#method.serve_connection
```

trailing period breaks the hyperlink

Reply:
Addressed in [a1e268c](https://github.com/DataDog/lading/pull/1187/commits/a1e268c2f29ecd894ada9f77e5744d7ac68451fa).